File: rye/src/main/java/com/rysonw/rye/Rye.java
package com.rysonwinterpreter.rye;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

public class Rye {
  public static void main(String[] args) throws IOException {
    System.out.println("Rye interpreter starting...");

    if (args.length > 1) {
        System.out.println("Usage: rye [script]");
        System.exit(64);
    } 
    else if (args.length == 1) {
        runFile(args[0]);
    } 
  }

  private static void runFile(string filePath) throws IOException {
    // Rye only runs code from files, no CLI
    byte[] bytes = Files.readAllBytes(Paths.get(filePath));
    runLines(bytes);
  }

  private static void runLines(String source) {
    Scanner scanner = new Scanner(source);
    List<Tokens> tokens = scanner.scanTokens();

    for (var token : tokens) {
        System.out.println(token);
    }
  }

  static void error(int line, String errorMessage) {
    report(line, "", errorMessage)
  }

  static void report() {
    System.error.println("[line " + line + "] Error" + where + ": " + message);
  }
}

File: rye/src/main/rye/Rye.cs
using System;
using System.IO;
using System.Text;
using System.Collections.Generic;
using com.rysonw.rye.frontend;

namespace com.rysonw.rye
{
    class Rye
    {
        static int Main(string[] args)
        {
            Console.WriteLine("Rye interpreter starting...");

            if (args.Length > 1)
            {
                Console.WriteLine("Usage: rye [script]");
                return 64;
            }
            else if (args.Length == 1)
            {
                try
                {
                    RunFile(args[0]);
                }
                catch (IOException ex)
                {
                    Console.Error.WriteLine(ex.Message);
                    return 1;
                }
            }
            else
            {
                Console.WriteLine("No script provided. Exiting.");
            }

            return 0;
        }

        private static void RunFile(string filePath)
        {
            var bytes = File.ReadAllBytes(filePath);
            var source = Encoding.UTF8.GetString(bytes);
            RunLines(source);
        }

        private static void RunLines(string source)
        {
            var scanner = new Scanner(source);
            List<Token> tokens = scanner.ScanTokens();

            foreach (var token in tokens)
            {
                Console.WriteLine(token);
            }
        }

        public static void Error(int line, string message)
        {
            Console.Error.WriteLine($"[line {line}] Error: {message}");
        }
    }
}
